OSDOCS#6669: Add OLM 1.0 arch/components (#65963)
openshift/openshift-docs: operators/olm_v1/arch/olmv1-components.adoc

:_content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="olmv1-components"]
= {olmv1} components overview (Technology Preview)
:context: olmv1-components

toc::[]

> :FeatureName: {olmv1}

Reviewer: :fire:

include::snippets/technology-preview.adoc[]

{olmv1-first} comprises the following component projects:

Operator Controller:: xref:../../../operators/olm_v1/arch/olmv1-operator-controller.adoc#olmv1-operator-controller[Operator Controller] is the central component of {olmv1} that extends Kubernetes with an API through which users can install and manage the lifecycle of Operators and extensions. It consumes information from each of the following components.

RukPak:: xref:../../../operators/olm_v1/arch/olmv1-rukpak.adoc#olmv1-rukpak[RukPak] is a pluggable solution for packaging and distributing cloud-native content. It supports advanced strategies for installation, updates, and policy.
+

> RukPak provides a content ecosystem for installing a variety of artifacts, such as Git repositories, Helm charts, and OLM bundles, on a Kubernetes cluster. The component can then manage, scale, and upgrade these artifacts in a safe way to enable powerful cluster extensions.

Reviewer: ```suggestion
RukPak provides a content ecosystem for installing a variety of artifacts on a Kubernetes cluster. Artifact examples include Git repositories, Helm charts, and OLM bundles.  The component can then manage, scale, and upgrade these artifacts in a safe way to enable powerful cluster extensions.
```

+

> At its core, RukPak is a small set of APIs, packaged as custom resource definitions (CRDs), and controllers that watch for those APIs. The APIs express content to install on a cluster and how to create a running deployment of the content.

Reviewer: ```suggestion
At its core, RukPak is a small set of APIs and controllers. The APIs are packaged as custom resource definitions (CRDs) that express what content to install on a cluster and how to create a running deployment of the content. The controllers watch for the APIs.
```
Maybe that changes the meaning too much, but I was getting a little tangled up in the original.

Author: > I was getting a little tangled up

Me too. Your change keeps my intended meaning/understanding, FWIW, will check during SME review to make sure still accurate.

Deppy:: xref:../../../operators/olm_v1/arch/olmv1-deppy.adoc#olmv1-deppy[Deppy] is a Kubernetes API for resolving constraints over catalogs of RukPak bundles. The initial goal of the project is to remove the dependency manager from the earlier OLM component and make it its own generic component.

> Catalogd:: xref:../../../operators/olm_v1/arch/olmv1-catalogd.adoc#olmv1-catalogd[Catalogd] is a Kubernetes extension that unpacks file-based catalog (FBC) content packaged and shipped in container images for consumption by clients on-clusters. As a component of the {olmv1} microservices architecture, catalogd hosts metadata for Kubernetes extensions packaged by the authors of the extensions, and as a result helps users discover installable content.

Reviewer: maybe, "on-cluster clients" or "clients on clusters".

Reviewer: If you pulled this from my docs, I should double check.
